### DISCUSSION TITLE
Formily 结合 MUI 实现动态表单遇到的问题

### DISCUSSION THREAD

**Discussion by daijingfeng** [alibaba/formily#3279] [Q&A 2.x] [answered]
例如 [codesandbox链接](https://codesandbox.io/s/formily-mui-react17-collapse-im4e44?file=/Form-1.tsx) 里的 <b>表单1</b>，是我们实际业务场景的例子。是想结合 Formily 和 [MUI](https://mui.com/zh/material-ui/react-text-field/) 来实现动态表单。里面有自定义模块、表单校验和联动交互。页面通过scheam描述出来了（暂时配合 @formily/antd里的Form和FormItem来用），但是有几个问题。

问题1：表单校验。必填字段输入内容后，提示仍然存在。另外其他校验也没有生效，比如 Desc 字段的字符长度限制，但在<b>表单2</b>中，必填字段和字符长度的校验都是正常的。
![image](https://user-images.githubusercontent.com/17313078/179207011-31f64d3e-058d-49d8-acba-bfe69993ebc8.png)

问题2：联动交互。想通过 Switch 来控制下面的自定义模块显示隐藏，但是没有触发交互。同样的写法在<b>表单2</b>，也是可以的。

![image](https://user-images.githubusercontent.com/17313078/179207165-7de24bc1-f1ef-4d5b-a258-cb300c7836b9.png)

麻烦大佬帮忙看看<b>表单1</b>是哪里的写法有问题。万分感谢！！

**Comment by janryWang** [comment]
看样子是你实现的Collapse自定义组件有问题，Name字段放到外面就行了

**Reply by daijingfeng** [reply]
感谢大佬！表单校验没问题了。但 switch 切换交互还是不行，我还没找到原因

**Comment by janryWang** [comment, marked as answer]
找到问题了，AccordionCollapse组件把props.onChange透传给div了，因为formily会给每个非void字段默认透传onChange，这样处理一下即可
```
 return <div {...props} onChange={()=>{}}>{renderItems()}</div>;
```

**Reply by weilaiqishi** [reply]
奇怪的知识增加了

**Reply by janryWang** [reply]
这没办法的，react组件div会收到冒泡事件
